4.5 release page (#1138)
godotengine/godot-website: collections/_release_4_5/entry-systems-animation-bind-bones-to-other-bones-with-boneconstraint3d.md

---
type: entry
section: systems
subsection: animation
rank: 0
importance: 2
anchor: bind-bones-to-other-bones-with-boneconstraint3d
title: Bind bones to other bones with `BoneConstraint3D`

> blockquote: yo dawg, I heard you like bones

Reviewer: ```suggestion
blockquote: Yo dawg, I heard you like bones
```

Author: I changed the sentence to follow more closely [the meme](https://knowyourmeme.com/memes/xzibit-yo-dawg) and the feature:

> Yo dawg, I heard you like constraints, so I put constraints in your bones so you can pose while you pose.

Reviewer: I've never heard that meme but OK!

text: |

>   With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransfromModifier3D`, and `ConvertTransfromModifier3D`, it is now possible to bind bones to other bones. It opens up the door for more natural movements and poses.

Reviewer: ```suggestion
  With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransformModifier3D`, and `ConvertTransformModifier3D`, it is now possible to bind bones to other bones. This opens up the door for more natural movements and poses.
```

>   With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransformModifier3D`, and `ConvertTransformModifier3D`, it is now possible to bind bones to other bones. This opens up the door for more natural movements and poses.

Reviewer 2: ```suggestion
  With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransformModifier3D`, and `ConvertTransformModifier3D`, it is now possible to bind bones to other bones. This opens the door for more natural movements and poses.
```

Author: Finally, I used "This can enable more natural movements and poses." (to not repeat the same "opens the door")

Reviewer 2: That should be "natural movement and poses", no plural, "movements" means something different (missed that one)

>   This feature is notably really helpful for handling virtual reality and metaverse avatars.

Reviewer 3: ```suggestion
  This feature is really helpful for VR and metaverse avatars.
```

---
type: entry
section: systems
subsection: animation
rank: 0
importance: 2
anchor: bind-bones-to-other-bones-with-boneconstraint3d
title: Bind bones to other bones with `BoneConstraint3D`
blockquote: Yo dawg, I heard you like constraints, so I put constraints in your 
  bones so you can move bones when bones move.
text: |

>   With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransformModifier3D`, and `ConvertTransformModifier3D`, it is now possible to bind bones to other bones. This can enable more natural movements and poses.

Reviewer 2: ```suggestion
  With `BoneConstraint3D` and the new `AimModifier3D`, `CopyTransformModifier3D`, and `ConvertTransformModifier3D`, it is now possible to bind bones to other bones. This can enable more natural movement and poses.
```
Plural here has a bit of a different meaning I'd say
